docs/src/features.md

```diff
@@ -182,11 +182,61 @@ Add these settings to LSP settings, your Sublime settings, Syntax-specific setti
 * `diagnostics_highlight_style` `"underline"` *highlight style of code diagnostics: "box", "underline", "stippled", "squiggly" or ""*
 * `highlight_active_signature_parameter`: *highlight the active parameter of the currently active signature*
 * `document_highlight_style`: *document highlight style: "box", "underline", "stippled", "squiggly" or ""*
-* `document_highlight_scopes`: *customize your sublime text scopes for document highlighting*
 * `diagnostics_gutter_marker` `"dot"` *gutter marker for code diagnostics: "dot", "circle", "bookmark", "sign" or ""*
 * `show_symbol_action_links` `false` *show links to symbol actions like go to, references and rename in the hover popup*
 * `disabled_capabilities`, `[]` *Turn off client capabilities (features): "hover", "completion", "documentHighlight", "colorProvider", "signatureHelp", "codeLensProvider", "codeActionProvider"*
 * `log_debug` `false` *show debug logging in the sublime console*
 * `log_server` `[]` *log communication from and to language servers*
 * `log_stderr` `false` *show language server stderr output in the console*
 * `log_max_size` `8192` *max  number of characters of payloads to print*
+
+### Color configurations
+
+Some features use TextMate scopes to control the colors (underline, background or text color) of styled regions in a document or popup.
+Colors can be customized by adding a rule for these scopes into your color scheme, see https://www.sublimetext.com/docs/3/color_schemes.html#customization.
+
+The following tables give an overview about the scope names used by LSP.
+
+#### Document Highlights
+
+!!! info "This feature is only available if the server has the *documentHighlightProvider* capability."
+    Highlights other occurrences of the symbol at a given cursor position.
+
+| scope | DocumentHighlightKind | description |
+| ----- | --------------------- | ----------- |
+| `markup.highlight.text.lsp` | Text | A textual occurrence |
+| `markup.highlight.read.lsp` | Read | Read-access of a symbol, like reading a variable |
+| `markup.highlight.write.lsp` | Write | Write-access of a symbol, like writing to a variable |
+
+If `document_highlight_style` is set to "fill" in the LSP settings, additionally `meta.fill.lsp` gets prepended to the listed scopes.
```

Comment:
Can you give an example on how one would achieve the effect as in your gif in the linked issue? I've checked out your branch, and these are my overridden rules for Mariana:

```js
{
    "rules": [
        {
            "scope": "markup.error",
            "foreground": "hsl(357, 79%, 65%)"
        },
        {
            "scope": "markup.warning",
            "foreground": "hsl(32, 93%, 66%)"
        },
        {
            "scope": "markup.info",
            "foreground": "hsl(0, 0%, 100%)"
        },
        {
            "scope": "markup.info.hint",
            "foreground": "hsl(0, 0%, 97%)"
        },
        {
            "scope": "markup.highlight",
            "background": "color(rgb(255, 255, 255) alpha(0.1))"
        }
    ],
}
```

It looks like this:

<img width="639" alt="image" src="https://user-images.githubusercontent.com/2431823/108118792-28b51880-709f-11eb-8b12-961225af5773.png">

It looks like the color of markup.changed and markup.inserted is still applied, but also together with my markup.highlight scope? I don't really understand what is going on here.

Reply:
This is correct, because `markup.changed`/`markup.inserted` will still match and they use (only) a foreground color. Therefore both your new rule for `markup.highlight` and the predefined rule from the color scheme for `markup.changed`/`markup.inserted` get applied. I think this is how color schemes work in general, it's just with the highest selector score gets applied last and then override potential colors from other rules for "background", "foreground" and "font_style".

What I did was setting both a foreground and background color in the rule for `markup.highlight` in my screenshot (with foreground using the default text color). You could see from my screenshot that the foreground color will even override the syntax highlighting for the function argument when I click on "alpha".

For Mariana it would be
```json
        {
            "scope": "markup.highlight",
            "foreground": "var(white3)",
            "background": "color(rgb(255, 255, 255) alpha(0.1))"
        }
```

In this PR I also added `meta.fill` to the scopes only when "fill" is used, so that color scheme authors could use a rule that applies only to filled regions, or tweak the style for both "fill" and other options.

Also I observed that setting an alpha value for the background color unfortunately has no effect for when the scope is used with `view.add_regions`. It would be nice to have blending colors for when "highlight_line" is used or within the red/green line backgrounds from ST's incremental diff feature.

Reply:
Also ST seems to alter its behavior how to draw `view.add_regions` withouth `sublime.DRAW_NO_FILL` dependent whether the used scope has a background color assigned or not. The default `markup.inserted/changed` scope does *not* have a background color, and in that case ST uses the foreground color as background and automatically adjusts the foreground to provide a sufficient contrast.

Reply:
It's even more confusing because ST seems to have another special behavior only when the scope name contains one of the `region` scopes. I replaced the `markup.inserted` and `markup.changed` with a region scope for testing (i.e. "region.greenish markup.highlight.read.lsp"), and in that case it looks like the original foreground color gets preserved, even with a color scheme rule with "foreground" for `markup.highlight`. And instead of the text foreground, this color scheme rule now controls the border color. Here is the result for Mariana with
```json
        {
            "scope": "markup.highlight",
            "foreground": "#f0f",
            "background": "color(rgb(255, 255, 255) alpha(0.1))"
        }
```
![mariana](https://user-images.githubusercontent.com/6579999/108129039-e810cb80-70ad-11eb-8e82-5438cd3acf7f.png)

And when I change the code to apply `sublime.DRAW_NO_OUTLINE` for "fill", the foreground color still has no effect:

![mariana2](https://user-images.githubusercontent.com/6579999/108129504-91f05800-70ae-11eb-8d30-6737807cd53b.png)

Perhaps replacing the `markup.changed/markup.inserted` scopes with `region` scopes and using the `sublime.DRAW_NO_OUTLINE` flag would be the best option to allow a nice highlighting style for "fill", while still providing a good default style for "underline" and the other options.

To be continued....